### ISSUE TITLE
Navigation: Accessibility: cannot reach sidebar navigation with a keyboard using tab focus

### ISSUE THREAD

**Issue by alisterscott** [Automattic/wp-calypso#694]
Originally raised by @designsimply 

Paraphrasing:

<blockquote>
1. All elements should have focus styles.<br>
2. Focused elements should be visible (you shouldn't be able to tab through hidden elements).<br>
3. Cannot tab into the main navigation because of Infinite scroll.
</blockquote>

Steps to reproduce:
1. Log in to WordPress.com
2. Press `Cmd+F5` or `Fn+Cmd+F5` to turn on VoiceOver on a Mac
3. Go to https://wordpress.com/
4. Try to find the main navigation by using the tab key.<br>Press `Ctl+Opt+Down` or `Ctl+Opt+Up` to enter or exit areas such as the toolbar.

<strong>Result:</strong> because of the way the elements are ordered, you must tab through all of the content before you can reach the main navigation. This makes it difficult to reach the main navigation on all Calypso pages but virtually impossible to reach the main navigation in the Reader because the infinite scroll area comes before the navigation area.

**Comment by amandarush** [comment]
Can also confirm same behavior on Windows with NVDA latest release and
Firefox latest release.







*From:* Alister Scott [mailto:notifications@github.com]
*Sent:* Tuesday, November 24, 2015 5:31 PM
*To:* Automattic/wp-calypso
*Subject:* [wp-calypso] Navigation: Accessibility: cannot reach sidebar
navigation with a keyboard using tab focus (#694)



Originally raised by @designsimply <https://github.com/designsimply>

Paraphrasing:

1. All elements should have focus styles.
2. Focused elements should be visible (you shouldn't be able to tab through
hidden elements).
3. Cannot tab into the main navigation because of Infinite scroll.

Steps to reproduce:

   1. Log in to WordPress.com
   2. Press Cmd+F5 or Fn+Cmd+F5 to turn on VoiceOver on a Mac
   3. Go to https://wordpress.com/
   4. Try to find the main navigation by using the tab key.
   Press Ctl+Opt+Down or Ctl+Opt+Up to enter or exit areas such as the
   toolbar.

*Result:* because of the way the elements are ordered, you must tab through
all of the content before you can reach the main navigation. This makes it
difficult to reach the main navigation on all Calypso pages but virtually
impossible to reach the main navigation in the Reader because the infinite
scroll area comes before the navigation area.

—
Reply to this email directly or view it on GitHub
<https://github.com/Automattic/wp-calypso/issues/694>.

**Comment by SiobhyB** [comment]
This is still an issue, changing from an "enhancement" to a "bug" as this prevents some people from being able to navigate WordPress.com.

**Comment by omidfi** [comment]
@alisterscott I sent a PR that addresses this: #22791

**Comment by omidfi** [comment]
This should be fixed now, via https://github.com/Automattic/wp-calypso/pull/22884
